### PR TITLE
Ecl 639 fixing how contact details are displayed in CYA page for amend registration

### PR DESCRIPTION
**Pull request by liamoc89**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyregistration/viewmodels/checkAnswers/CheckYourAnswersViewModel.scala [hunk 1]
```diff
@@ -214,16 +214,16 @@ case class CheckYourAnswersViewModel(
           hasSecondContactDetailsPresentChanged,
           SecondContactSummary.row(registration.contacts.secondContact)
         ) ++ addIf(
-          hasSecondContactNameChanged,
+          hasSecondContactNameChanged && hasSecondContact,
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyregistration/viewmodels/checkAnswers/CheckYourAnswersViewModel.scala:217]: Make this it's own appropriately named viewModel property/val

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/AmendRegistrationPdfView.scala.html [hunk 2]
```diff
@@ -38,7 +38,7 @@ <h2>@messages("checkYourAnswers.organisationDetails")</h2>
     @govukSummaryList(viewModel.organisationDetails())
 
 
-    @if(viewModel.hasSecondContact) {
+    @if(viewModel.hasSecondContact && !viewModel.hasSecondContactDetailsPresentChanged) {
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyregistration/views/AmendRegistrationPdfView.scala.html:41]: Make this it's own appropriately named viewModel property/val

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/CheckYourAnswersView.scala.html [hunk 3]
```diff
@@ -55,7 +55,7 @@ <h2 class="govuk-heading-m">@messages("checkYourAnswers.organisationDetails")</h
             @govukSummaryList(viewModel.organisationDetails())
         }
 
-        @if(viewModel.hasSecondContact) {
+        @if(viewModel.hasSecondContact && !viewModel.hasSecondContactDetailsPresentChanged) {
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyregistration/views/CheckYourAnswersView.scala.html:58]: Make this it's own appropriately named viewModel property/val

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/AmendRegistrationPdfView.scala.html (hunk 2 above)
Comment (liamoc89) [reply app/uk/gov/hmrc/economiccrimelevyregistration/views/AmendRegistrationPdfView.scala.html:41]: hasSecondContactDetailsPresentChanged is already it's own val in TrackRegistrationChanges and is unit tested

File: app/uk/gov/hmrc/economiccrimelevyregistration/views/CheckYourAnswersView.scala.html (hunk 3 above)
Comment (liamoc89) [reply app/uk/gov/hmrc/economiccrimelevyregistration/views/CheckYourAnswersView.scala.html:58]: hasSecondContactDetailsPresentChanged is already it's own val in TrackRegistrationChanges and is unit tested

File: app/uk/gov/hmrc/economiccrimelevyregistration/viewmodels/checkAnswers/CheckYourAnswersViewModel.scala (hunk 1 above)
Comment (liamoc89) [reply app/uk/gov/hmrc/economiccrimelevyregistration/viewmodels/checkAnswers/CheckYourAnswersViewModel.scala:217]: Should this be in TrackRegistrationChanges? As it's not necessarily a change to the registration, it's just confirming if the secondContact is present

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**